workshop/content/basics/templates/pseudo-parameters/_index.md

```diff
@@ -0,0 +1,238 @@
+
+![lambda-test](pseudo-parameters/lambda-test.png)
+After you invoke the Lambda function, under the **Function Logs** section you should see output similar to the example shown next:
+
+![lambda-png](pseudo-parameters/lambda.png)
+
+You should see a line, similar to the one above, showing `alice` as the value for your `dbUsername` parameter: with this, you have verified that the logic you added for your Lambda function can access your parameter, and retrieve and print its value as you expected.
+
+
+### Challenge
+You’ve learned how to use pseudo parameters in your CloudFormation templates. Now let’s say you want to add an [Amazon S3 bucket](https://aws.amazon.com/s3/) to your CloudFormation template: for example, you choose the name of the S3 bucket to be as in the following format: `YOUR_BUCKET_NAME_PREFIX-AWS_REGION-YOUR_ACCOUNT_ID`, such as: `my-demo-bucket-us-east-1-111122223333`.
+
+**Task:** Describe an S3 bucket resource in your template. Add a prefix for your bucket name, and pass this prefix in as a CloudFormation template parameter. Use your template parameter and pseudo parameters to compose the bucket name as in the format mentioned earlier.
+
+{{%expand "Need a hint?" %}}
+
+- See the documentation for [CloudFormation parameters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html), and define a parameter for your S3 bucket name prefix in your template.
+- When composing the bucket name, reference your CloudFormation parameter like you referenced pseudo parameters with the `!Sub` intrinsic function earlier. For example, if your template parameter is `S3BucketNamePrefix`, choose to reference it with the `!Sub` intrinsic function as follows: `!Sub '${S3BucketNamePrefix}'`
+
+{{% /expand %}}
+
+
+
+{{%expand "Want to see the solution?" %}}
+```YAML
+# Add a template parameter that will let users input a bucket name prefix
+Parameters:
+  S3BucketNamePrefix:
+    AllowedPattern: ^[0-9a-zA-Z]+([0-9a-zA-Z-]*[0-9a-zA-Z])*$
+    ConstraintDescription: Bucket name prefix can include numbers, lowercase
+      letters, uppercase letters, and hyphens (-). It cannot start or end with a hyphen
+      (-).
+    Description: The prefix to use for your S3 bucket
+    Type: String
+    MinLength: 3
+    Default: my-demo-bucket
+
+Resources:
+  DemoBucket:
+      Type: 'AWS::S3::Bucket'
```

Comment:
Remove single quotes for the value for `Type` here and in template files for this lab, for consistency.